docs/synapse/userguides/background.rst

```diff
@@ -0,0 +1,208 @@
+
+.. _bkd-graphs:
+
+Graphs
+------
+
+A **graph** is a mathematical structure used to model pairwise relations between objects. Graphs consist of:
+
+- **vertices** (or **nodes**) that represent objects, and
+- **edges** that connect two vertices in some type of relationship.
+
+|graph|
+
+Edges connect exactly two nodes; they are "pairwise" or "two-dimensional". Both nodes and edges may have properties
+that describe their relevant features. In this sense both nodes and edges can be thought of as representational
+objects within the graph: nodes typically represent things ("nouns") and edges typically represent relationships
+("verbs").
+
+**Examples**
+
+**Cities and Roads.** A simple example of data that can be represented by a graph are cities connected by
+roads. If abstracted into graph format, each city would be a vertex or node and a road connecting two cities
+would be an edge. Since you can travel from City A to City B or from City B to City A on the same road, the
+graph is **directionless** or **undirected.**
+
+**Social Networks.** Another example is social networks based on "connections", such as LinkedIn. In this case,
+each person would be a node and the connection between two people would be an edge. In most cases, LinkedIn
+requires a mutual connection (you must request a connection and the other party must accept); in this sense it can
+be considered a directionless graph. (This is a simplification, but serves our purpose as an example.)
+
+.. _bkd-directed-graphs:
+
+Directed Graphs
+---------------
+
+A **directed graph** is a graph where the edges have a direction associated with them. In other words, the
+relationship represented by the edge is one-way. Where an edge in an undirected graph is often represented by
+a straight line, an edge in a directed graph is represented by an arrow.
+
+|directedgraph|
+
+**Examples**
+
+**Cities and Roads.** In our cities-and-roads example, the graph would be a directed graph if the roads were
+all one-way streets: in this case you can use a particular road to go from City A to City B, but not from City
+B to City A.
+
+**Social Networks.** Social networks that support a "follows" relationship (such as Twitter) can be represented
+as directed graphs. Each person is still a node, but the "follows" relationship is one way – I can "follow" you,
+but you don’t have to follow me. If you choose to follow me, that would be a second, independent one-way edge
+in the opposite direction. (This is also a simplification but works for a basic illustration.)
+
+**Other Examples.** Many other types of data can be represented with nodes and directed edges.  For example, in
+information security you can represent data and relationships such as:
+
+``malware_file --(performed DNS lookup for)--> domain``
+
+or
+
+``domain --(resolves to)--> ip_address``
+
+In these examples, files, domains and IP addresses are nodes and "performed DNS lookup for" and "resolves to"
```

Comment:
oxford comma after domains

Reply:
Gaaah this is a firing offense at Vertex :scream: